lib/policies/oauth2/site.js

```diff
@@ -1,13 +1,25 @@
-'use strict';
-
 const passport = require('passport');
 const path = require('path');
+const services = require('../../services/index');
 
 module.exports.loginForm = (request, response) => response.render(path.join(__dirname, 'views/login'));
 
 module.exports.login = passport.authenticate('local', { successReturnToOrRedirect: '/', failureRedirect: '/login' });
 
-module.exports.logout = (request, response) => {
-  request.logout();
-  response.redirect('/');
+module.exports.logout = (req, res) => {
+  req.logout();
+  if (!req.query.returnTo) {
+    return res.redirect('/'); // TODO: implement settings section for default view path and return redirects
+  }
+  const returnUrl = decodeURI(req.query.returnTo);
```

Comment:
Triple check: I think that express is already decoding the parameters for us under the hood.